hazelcast/src/main/java/com/hazelcast/config/AbstractWanPublisherConfig.java

```diff
@@ -155,16 +161,16 @@ public boolean equals(Object o) {
 
         AbstractWanPublisherConfig that = (AbstractWanPublisherConfig) o;
 
-        if (!publisherId.equals(that.publisherId)) {
-            return false;
-        }
-        if (!Objects.equals(className, that.className)) {
-            return false;
-        }
-        if (!Objects.equals(implementation, that.implementation)) {
-            return false;
+        return publisherId.equals(that.publisherId)
+            && Objects.equals(replicationPublisherNameInternal(), that.replicationPublisherNameInternal())
+            && properties.equals(that.properties);
+    }
+
+    private String replicationPublisherNameInternal() {
```

Comment:
I agree with you. It should also simplify code in the `ConfigChecker`s that are used for config testing.

Thank you for your thorough reviews.

Reply:
> It should also simplify code in the `ConfigChecker`s that are used for config testing.

Nope 😄 visa versa. We have to introduce functionality to check configs, because after serialisation deserialisation `Config`s will contain `className`s instead of `implementation`s